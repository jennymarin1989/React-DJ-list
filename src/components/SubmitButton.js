import React from 'react';

const SubmitButton = (props) => {
    
    return(
        <button id={props.id} className="btn btn-success" onClick={props.handleClick}>{props.name}</button>
    )

}

export default SubmitButton;