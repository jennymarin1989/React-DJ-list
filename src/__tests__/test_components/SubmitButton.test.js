import React from 'react';
import { shallow } from 'enzyme';
import SubmitButton from '../../components/SubmitButton';

describe ("SubmitButton", () => {
    
    let button = shallow(<SubmitButton/>)
  
    it("renders correctly", () => {
       expect(button).toMatchSnapshot(); 
    });

    it("renders button correctly", ()=>{
      expect(button.getElement('button')).toBeDefined();
    })

    // it("handles clicks", () =>{
    //   let OnClickSpy = jest.fn();
    //   let listByYear = shallow(<SubmitButton  onClick={OnClickSpy}/>);
    //   listByYear.find('button').simulate('click');
    //   expect(OnClickSpy).toHaveBeenCalled();
    // })

  })
