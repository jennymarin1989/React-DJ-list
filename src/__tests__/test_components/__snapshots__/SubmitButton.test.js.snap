// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SubmitButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "btn btn-success",
      "id": undefined,
      "onClick": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "btn btn-success",
        "id": undefined,
        "onClick": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
